src/common/marlin_server.hpp

```diff
@@ -27,22 +29,21 @@ void fsm_change(ClientFSM type, T phase, fsm::PhaseData data) {
 class ClientResponseHandler : public ClientResponses {
     ClientResponseHandler() = delete;
     ClientResponseHandler(ClientResponseHandler &) = delete;
-    static uint32_t server_side_encoded_response;
+    static std::optional<uint32_t> server_side_encoded_response;
 
 public:
     //call inside marlin server on received response from client
     static void SetResponse(uint32_t encoded_bt) {
         server_side_encoded_response = encoded_bt;
     }
     //return response and erase it
-    //return -1 if button does not match
     template <class T>
     static Response GetResponseFromPhase(T phase) {
-        uint32_t _phase = server_side_encoded_response >> RESPONSE_BITS;
+        uint32_t _phase = server_side_encoded_response.value_or(-1) >> RESPONSE_BITS;
```

Comment:
Again, the point of optional is not to have these special values and going back to them is kind of against that. Is it possible to restructure this function? Maybe first check there is something and if not, return Response::_none right away?